### PR TITLE
Remove timeout, code cleanup

### PR DESCRIPTION
**Pull request by JosefWN**
Proposal for https://github.com/shamblett/coap/issues/78

- Now only relying on specified timeouts in `CoapConfig`
- Requests return null response on timeout
- The utils actually made it more complicated to manipulate lists, so I removed those utils and shortened the code a bit (I find the standard library easier to read than custom utils, too)
- Changed some nullables to non-nullables
- Removed some unused code

You can still ignore an ongoing request (equivalent to the previous timeout):

```dart
var respFuture = client.get('test');
respFuture.then((CoapResponse? resp) => ...);
// ...
respFuture.ignore();
```

@JKRhb @shamblett thoughts?

### CODE REVIEW

**Review by JKRhb — APPROVED** [review]
Thank you, @JosefWN, I think this is another great improvement :)

File: lib/src/util/coap_util.dart [hunk 1]
```diff
@@ -7,88 +7,8 @@
...
-      if (item != source.last) {
-        sb.write(',');
-      }
-    }
-    return sb.toString();
-  }
-
   /// Stringify options in a message.
   static String optionsToString(CoapMessage msg) {
     final sb = StringBuffer();
```
Comment (JKRhb) [inline lib/src/util/coap_util.dart:14]: On second glance, maybe this method could also be moved to the `CoapMessage` class? If I see it correctly, that is the only place where the method is being used.

File: lib/src/util/coap_util.dart [hunk 2]
```diff
@@ -7,88 +7,8 @@
...
-
-  /// The timer callback
-  void _timerCallback() {
-    _running = false;
-    _completer.complete();
-  }
-}
-
 /// Utility methods
 class CoapUtil {
```
Comment (JKRhb) [inline lib/src/util/coap_util.dart:11]: I think in this class, you could also remove the `regIP` member and use `InternetAddress.tryParse(host) == null` to perform the IP address check.

Comment (JKRhb) [reply lib/src/util/coap_util.dart:11]: The remaining `lookupHost` method could then be moved to the `CoapClient` class.

File: lib/src/util/coap_util.dart (hunk 1 above)
Comment (JosefWN) [reply lib/src/util/coap_util.dart]: Nice catch, fixed: https://github.com/shamblett/coap/pull/87/commits/7136515b45582fce34bce0c8746722ae2ae23f81

**Comment by shamblett** [comment]
OK guys happy with this, I'll merge rt if you're ready.

**Comment by JosefWN** [comment]
Added some minor tweaks for cancelled requests and pings (since token is empty I'm matching on id instead): https://github.com/shamblett/coap/pull/87/commits/280d636b7404c697696d1120407affc13eb81f84.

Ready for merge! :)

**Pull request merged**